src/vcpkg/paragraphs.cpp

```diff
@@ -99,9 +99,9 @@ namespace vcpkg
         return result;
     }
 
-    static Optional<std::pair<std::string, TextRowCol>> remove_field(Paragraph* fields, const std::string& fieldname)
+    static Optional<std::pair<std::string, TextRowCol>> remove_field(Paragraph* fields, StringView fieldname)
     {
-        auto it = fields->find(fieldname);
+        auto it = fields->find(fieldname.to_string());
```

Comment:
This is now safe but I think whether it's a perf win is questionable, since a lot of the time `fieldname` is already going to have been a `std::string`. However, I think the *interface* change is an improvement so I'm OK with it. Engaging C++17 transparent map comparisons can be later.

Reply:
> This is now safe but I think whether it's a perf win is questionable, since a lot of the time `fieldname` is already going to have been a `std::string`.

Even if it was a win, anybody would notice because there are other things that are much slower. I'm already working on a PR to speed up post build lint using concurrency for some checks.